sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -264,31 +264,44 @@ public BulkExecutor(CosmosAsyncContainer container,
 
             // First check if it failed due to split, so the operations need to go in a different pk range group. So
             // add it in the mainSink.
-            if (cosmosException.getStatusCode() == HttpResponseStatus.GONE.code() &&
-                itemBulkOperation.getRetryPolicy().shouldRetryForGone(
-                    cosmosException.getStatusCode(),
-                    cosmosException.getSubStatusCode())) {
+            if (cosmosException.getStatusCode() == HttpResponseStatus.GONE.code()) {
```

Comment:
should this `if` be removed? since the code already calls `retryOtherExceptions` if the policy didn't handle the current one?

Reply:
Have that if for early branching out. But removed it now as it makes code more cleaner